fairscale/utils/testing.py

```diff
@@ -82,7 +82,7 @@ def torch_version() -> Tuple[int, ...]:
     return tuple(int(n) for n in numbering)
 
 
-def dist_init(rank: int, world_size: int, filename: str) -> None:
+def dist_init(rank: int, world_size: int, filename: str) -> bool:
     """
     Initialize torch distributed, based on a temporary file shared across ranks, which makes it possible for unrelated
     tests to be run concurrently.
```

Comment:
I'd add here:

Return False if not enough GPUs present in the system. We can't use pytest.skip here since this is in a forked process.